### PR TITLE
Add note about improvement to config cache dependency resolution

### PR DESCRIPTION
**Pull request by bamboo**
See #21252

### CODE REVIEW

File: subprojects/docs/src/docs/release/notes.md [hunk 1]
```diff
@@ -147,6 +147,12 @@ See the [documentation](userguide/command_line_interface.html#sec:builtin_task_o
 
 The [configuration cache](userguide/configuration_cache.html) improves build time by caching the result of the configuration phase and reusing this for subsequent builds.
 
+#### Recovering from dependency resolution failures
+
+In previous Gradle versions it was possible to leave a configuration cache entry in a permanently broken state after a dependency resolution failure, even though the same build would later succeed with configuration caching disabled.
+
+Starting with Gradle 7.6 this is no longer the case and Gradle will be able to recover from dependency resolution failures in exactly the same way regardless of whether configuration caching is enabled or not.
```
Comment (nathan-contino) [inline subprojects/docs/src/docs/release/notes.md:154]:
```suggestion
#### Improved configuration cache failure recovery

In previous Gradle versions, it was possible to leave a configuration cache entry in a permanently broken state after a dependency resolution failure. The same build would later succeed with configuration caching disabled.

Starting with Gradle 7.6 this is no longer the case. Gradle recovers from dependency resolution failures in exactly the same way with the configuration cache enabled.
```

Simplified sentences and made the section title a little easier to parse and connect with a concrete improvement.

**Review by nathan-contino — APPROVED** [review]
LGTM, suggested a small wording improvement.

**Comment by bamboo** [comment]
Thanks, @nathan-contino!

**Comment by bamboo** [comment]
@bot-gradle test and merge

**Pull request merged**